tests/samplers_tests/test_nsgaii.py

```diff
@@ -146,6 +148,151 @@ def constraints_func(trial: FrozenTrial) -> Sequence[float]:
+def test_constrained_dominates_feasible_vs_infeasible(
+    direction: StudyDirection,
+) -> None:
+
+    # Check all pairs of trials consisting of these constraint values.
+    constraints_1d_feasible = [-float("inf"), -1, 0]
+    constraints_1d_infeasible = [float("nan"), 2, float("inf")]
+
+    directions = [direction]
+
+    # Feasible constraints.
+    constraints_list1 = [
+        [c1, c2] for c1 in constraints_1d_feasible for c2 in constraints_1d_feasible
+    ]
+    # Infeasible constraints.
+    constraints_list2 = [
+        [c1, c2]
+        for c1 in constraints_1d_feasible + constraints_1d_infeasible
+        for c2 in constraints_1d_infeasible
+    ]
+
+    # In the following code, we test that the feasible trials always dominate
+    # the infeasible trials.
+    for constraints1 in constraints_list1:
+        for constraints2 in constraints_list2:
+            t1 = _create_frozen_trial(0, [0], constraints1)
+            t2 = _create_frozen_trial(1, [1], constraints2)
+            assert _constrained_dominates(t1, t2, directions)
+            assert not _constrained_dominates(t2, t1, directions)
+
+            t1 = _create_frozen_trial(0, [1], constraints1)
+            t2 = _create_frozen_trial(1, [0], constraints2)
+            assert _constrained_dominates(t1, t2, directions)
+            assert not _constrained_dominates(t2, t1, directions)
+
+
+@pytest.mark.parametrize("direction", [StudyDirection.MINIMIZE, StudyDirection.MAXIMIZE])
+def test_constrained_dominates_infeasible_vs_infeasible(direction: StudyDirection) -> None:
+    inf = float("inf")
+    directions = [direction]
+
+    # The following table illustrates the violations of some constraint values.
+    # When both trials are infeasible, the trial with smaller violation dominates
+    # the one with larger violation.
+    #
+    #                       c[1]
+    #      ╔═════╤═════╤═════╤═════╤═════╤═════╗
+    #      ║     │ -1  │  0  │  1  │  2  │  ∞  ║
+    #      ╟─────┼─────┼─────┼─────┼─────┼─────╢
+    #      ║ -1  │           │  1  │  2  │  ∞  ║
+    #      ╟─────┼─feasible ─┼─────┼─────┼─────╢
+    #      ║  0  │           │  1  │  2  │  ∞  ║
+    # c[0] ╟─────┼─────┼─────┼─────┼─────┼─────╢
+    #      ║  1  │  1  │  1  │  2  │  3  │  ∞  ║
+    #      ╟─────┼─────┼─────┼─────┼─────┼─────╢
+    #      ║  2  │  2  │  2  │  3  │  4  │  ∞  ║
+    #      ╟─────┼─────┼─────┼─────┼─────┼─────╢
+    #      ║  ∞  │  ∞  │  ∞  │  ∞  │  ∞  │  ∞  ║
+    #      ╚═════╧═════╧═════╧═════╧═════╧═════╝
+    #
+
+    # Check all pairs of these constraints.
+    constraints_infeasible_sorted: List[List[List[float]]]
+    constraints_infeasible_sorted = [
+        # violation 1
+        [[1, -inf], [1, -1], [1, 0], [0, 1], [-1, 1], [-inf, 1]],
+        # violation 2
+        [[2, -inf], [2, -1], [2, 0], [1, 1], [0, 2], [-1, 2], [-inf, 2]],
+        # violation 3
+        [[3, -inf], [3, -1], [3, 0], [2, 1], [1, 2], [0, 3], [-1, 3], [-inf, 3]],
+        # violation inf
```

Comment:
Could you follow the coding style of the code comments?
https://github.com/optuna/optuna/wiki/Coding-Style-Conventions#block-and-inline-comments-are-supposed-to-be-sentences

Reply:
Thanks, I change it to the following.
```suggestion
        # These constraints have violation 1.
        [[1, -inf], [1, -1], [1, 0], [0, 1], [-1, 1], [-inf, 1]],
        # These constraints have violation 2.
        [[2, -inf], [2, -1], [2, 0], [1, 1], [0, 2], [-1, 2], [-inf, 2]],
        # These constraints have violation 3.
        [[3, -inf], [3, -1], [3, 0], [2, 1], [1, 2], [0, 3], [-1, 3], [-inf, 3]],
        # These constraints have violation inf.
```